### PR TITLE
Fixes #1702. wildcards_A01_t01.dart. Use correct types in list pattern

### LINKED ISSUE #1702: co19/LanguageFeatures/Patterns/wildcards_A01_t01
```
FAILED: dart2analyzer-none release_arm64 co19/LanguageFeatures/Patterns/wildcards_A01_t01
Expected: Pass
Actual: CompileTimeError

--- Command "dart2analyzer" (took 06.000101s):
DART_CONFIGURATION=ReleaseARM64 sdk/bin/dart --packages=.dart_tool/package_config.json pkg/analyzer_cli/bin/analyzer.dart --use-analysis-driver-memory-byte-store --dart-sdk=/Users/scheglov/Source/Dart/sdk.git/sdk/sdk -Dtest_runner.configuration=custom-configuration-1 --enable-experiment=patterns --ignore-unrecognized-flags --packages=/Users/scheglov/Source/Dart/sdk.git/sdk/.dart_tool/package_config.json --format=json /Users/scheglov/Source/Dart/sdk.git/sdk/tests/co19/src/LanguageFeatures/Patterns/wildcards_A01_t01.dart

unexpected analysis errors in wildcards_A01_t01.dart:
- Line 65, column 8: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'String'.

- Line 65, column 26: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'num'.

- Line 70, column 10: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'int'.

- Line 70, column 23: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'num'.

- Line 73, column 13: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'String'.

- Line 73, column 26: COMPILE_TIME_ERROR.PATTERN_TYPE_MISMATCH_IN_IRREFUTABLE_CONTEXT
  The matched value of type 'Object' isn't assignable to the required type 'num'.

--- Re-run this test:
python3 tools/test.py -m release -c dart2analyzer -a arm64 co19/LanguageFeatures/Patterns/wildcards_A01_t01
```

Code
```dart
  var list2 = ["1", 2, "3", 4];
  var [String _, two, _, num ___] = list2;
```

1. `num ___` is not a wildcard.
2. The required type of `[String _, two, _, num ___]` is `GLB(String, num) = Object`, and `Object` cannot be assigned to `String _`.

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Patterns/wildcards_A01_t01.dart [hunk 1]
```diff
@@ -61,16 +61,16 @@ main() {
...
   Expect.equals(2, two);
   Expect.equals(4, ___);
 
   Expect.throws(() {
-    var [int _, _, _, num _] = list2;
+    var [int _, _, int _, num _] = list2;
   });
   Expect.throws(() {
-    var [_, String _, _, num _] = list2;
+    var [_, num _, _, num _] = list2;
```
Comment (eernstg) [inline LanguageFeatures/Patterns/wildcards_A01_t01.dart:73]: Not sure why this one would throw, what am I missing?

**Review by eernstg — COMMENTED** [review]
One question came up

File: LanguageFeatures/Patterns/wildcards_A01_t01.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Patterns/wildcards_A01_t01.dart]: Indeed. Removed this check. The one above is enough. Thank you

File: LanguageFeatures/Patterns/wildcards_A01_t01.dart [hunk 2]
```diff
@@ -61,16 +61,13 @@ main() {
   Expect.equals(2, _two);
   Expect.equals(4, __);
 
-  var list2 = ["1", 2, "3", 4];
-  var [String _, two, _, num ___] = list2;
+  var list2 = [1, 2, 3.14, 4];
+  var [int _, two, _, num ___] = list2;
```
Comment (eernstg) [inline LanguageFeatures/Patterns/wildcards_A01_t01.dart:65]: I'd expect this to be a compile-time error, again because of the built-in downcast.

**Review by eernstg — COMMENTED** [review]
The downcast-by-variablePattern issue came up again.

**Comment by sgrekhov** [comment]
Hope now it is fiexd :) Please rereview

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**